domain/lib/src/repository/document/document_repository.dart

```diff
@@ -60,9 +60,19 @@ abstract class DocumentRepository {
 
   Future<List<Document>> copyToMySpace(CopyRequest copyRequest);
 
-  Future<String> downloadPreviewDocument(Document document, DownloadPreviewType downloadPreviewType, Token token, Uri baseUrl, CancelToken cancelToken);
+  Future<String> downloadPreviewDocument(Document document, DownloadPreviewType downloadPreviewType, Token token, Uri baseUrl, CancelToken cancelToken, {DownloadType downloadType = DownloadType.PREVIEW});
 
   Future<DocumentDetails> getDocument(DocumentId documentId);
 
   Future<Document> editDescription(DocumentId documentId, EditDescriptionDocumentRequest request);
+
+  Future<Document> isAvailableOffline(DocumentId documentId);
+
+  Future<bool> updateDocumentOffline(DocumentOffline documentOffline);
+
+  Future<bool> makeAvailableOffline(DocumentOffline documentOffline);
+
+  Future<bool> disableAvailableOffline(DocumentOffline documentOffline);
```

Comment:
why not the input is `DocumentId`?

Reply:
We need more information than DocumentID

Reply:
How we can provide `DocumentOffline` for this API. And what is your `more informatiopn`?.